src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/service/PrisonService.kt

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.digital.hmpps.welcometoprison.service
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.welcometoprison.client.PrisonApiClient
+import uk.gov.justice.digital.hmpps.welcometoprison.exception.NotFoundException
+import uk.gov.justice.digital.hmpps.welcometoprison.model.prison.PrisonerImage
+
+@Service
+class PrisonService(@Autowired private val client: PrisonApiClient) {
+
+  fun getPrisonerImage(offenderNumber: String): ByteArray? {
+    val prisonerImage: PrisonerImage? =
+      client.getPrisonerImages(offenderNumber)
+        .filter { it.imageView.equals("FACE") && it.imageOrientation.equals("FRONT") }
+        .maxByOrNull { it.captureDate }
+
+    if (prisonerImage === null) throw NotFoundException("No front-facing image of the offenders face found for offender number: $offenderNumber")
```

Comment:
Thought I'd go for a bit of logging here, see if we can track how often we don't have a front-facing photo of an offenders face and then decide if we might need to tweak the logic later? 🤔 Maybe be okay with a side-facing photo instead?

Reply:
I couldn't see much within the prison-api about what actually might be returned here, might need some NOMIS expertise input

Reply:
You can use some nice Kotli null handling n here - `return prisonerImage?.let { client.getPrisonerImage(it.imageId) } ?: throw NotFoundException(etc)`